tests/loops/test_loop_state_dict.py

```diff
@@ -37,6 +39,7 @@ def test_loops_state_dict():
     assert fit_loop.state_dict() == new_fit_loop.state_dict()
 
 
+@mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"})
```

Comment:
it's confusing. Above, you made the change to save the state dict by default yet here the test gets constrained to fault-tolerant training.

Reply:
This is a unit test for the state dict. Since I moved the fault-tolerant enabled checks inside of it, we need to make sure it's enabled